clients/app/src/views/components/compositions/FanClubInfo/FanClubInfo.tsx

```diff
@@ -11,12 +11,39 @@ import { getReadableNumber } from 'utils/helpers/readableNumber';
 import preventMultiCalls from 'utils/helpers/preventMultiCalls';
 import { ProfileListTypeEnum } from 'views/screens/ProfileList';
 import { IFanClubInfoProps } from './types';
+import buildFanClubMembersStore from 'stores/fanClubMembers';
 
 const FanClubInfo: React.FC<IFanClubInfoProps> = ({ data }) => {
   const { pushTo } = useNavigationWithParams();
   const { colors } = useTheme();
 
-  const isJoined = false;
+  const fanClubMembersStore = React.useMemo(() => buildFanClubMembersStore(), []);
```

Comment:
Is not it better to move outside of Component?

Reply:
Then we need to reset the state to default on component unmount, otherwise it will use the previous state status for a new one, e.g.

if in the previous case joining to the fan club was failed, then in the new one status will be initialized as `error`.

I used memo to initialize the state from zero for each one.